lib/backend/test/suite.go

```diff
@@ -21,90 +21,212 @@ package test
 import (
 	"context"
 	"encoding/hex"
+	"errors"
 	"math/rand"
+	"sync"
 	"sync/atomic"
+	"testing"
 	"time"
 
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/lib/backend"
-	"github.com/gravitational/teleport/lib/fixtures"
 
 	"github.com/gravitational/trace"
 
 	"github.com/jonboulle/clockwork"
 	"github.com/pborman/uuid"
 	"github.com/stretchr/testify/require"
-	"gopkg.in/check.v1"
 )
 
-type BackendSuite struct {
-	B backend.Backend
-	// B2 is a backend opened to the same database,
-	// used for concurrent operations tests
-	B2         backend.Backend
-	NewBackend func() (backend.Backend, error)
-	Clock      clockwork.FakeClock
+var (
+	ErrMirrorNotSupported           = errors.New("mirror mode not supported")
+	ErrConcurrentAccessNotSupported = errors.New("concurrent access not supported")
+)
+
+type ConstructionOptions struct {
+	MirrorMode bool
+
+	// ConcurrentBackend indicates that the Backend Constructor function should not
+	// create an entirely independent data store, but instead should create a
+	// new interface to the same underlying data store as `ConcurrentBackend`.
+	ConcurrentBackend backend.Backend
+}
+
+// ApplyOptions constructs a new `ConstructionOptions` value from a
+// sensible default and then applies the supplied options to it.
+func ApplyOptions(options []ConstructionOption) (*ConstructionOptions, error) {
+	result := ConstructionOptions{
+		MirrorMode: false,
+	}
+	if err := result.Apply(options); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// applyOptions applies a collection of option-setting functions to the
```

Comment:
Nit:
```suggestion
// Apply applies a collection of option-setting functions to the
```

Reply:
or maybe you want to actually rename this back to `applyOptions` so it mimics the public function it's being called in